python/ray/tests/test_advanced_4.py

```diff
@@ -96,6 +97,38 @@ def test_jemalloc_env_var_propagate():
     assert actual == expected
 
 
+def test_back_pressure():
+    ray.init()
+
+    signal_actor = Semaphore.options(max_pending_calls=10).remote(value=0)
+
+    try:
+        for i in range(10):
+            signal_actor.acquire.remote()
+    except ray.exceptions.BackPressureError:
+        assert False
+
+    with pytest.raises(ray.exceptions.BackPressureError):
```

Comment:
Imagine this scenario
max pending = 2

Submission order: task A, B, C, D, E, F, G

Is backpressure error guaranteed to be returned from the correct order? For example, A and B will be pending, and then is it guaranteed that C will raise an exception?

Reply:
Yes, C will raise an exception if A and B is still pending when C submitted.

Reply:
How is it guaranteed now? Are we always sending tasks in the submission order from the caller side?

Reply:
Yeah, that's it. The length of pending queue is limited to 2, A and B makes the queue full, C can not be pended then backpressure error returned. It is quite simple.

Reply:
Let me read the impl again before confirming this. One thing to he careful is we currently has a guarantee that caller order is guaranteed on the callee side. I feel like it can be broken in this kind of scenario

A, B, C, D with limit 2

A, B received
C rejected
A done
D processed

this will break our ordering guarantee. Let me read code again to see if this is possible

Reply:
When the user submit C, backpressure exception raised, the user should catch the exception and resubmit C before submit D. The design just limit the size of client queue of caller side.

Reply:
So backpressure is decided from the caller side?

Reply:
Hmm this is interesting. So we have backpressure limit per caller in this way right? So, if we have 10 callers, the backpressure limit would be 10 X backpressure limit per caller. Am I correct? 

I originally thought we have backpressure limit per actor

Reply:
Yeah, that's it. We think it's more simpler than limiting per actor, and we haven't found any problems so far.

Reply:
Yeah makes sense. I like this better as well.